src/modal/modal.component.ts

```diff
@@ -132,20 +155,24 @@ export class Modal implements AfterViewInit {
 	 */
 	constructor(public modalService: ModalService) {}
 
+	ngOnChanges({ open }: SimpleChanges) {
+		if (open && open.currentValue) {
+			// `100` is just enough time to allow the modal
+			// to become visible, so that we can set focus
+			setTimeout(() => this.focusInitialElement(), 100);
+		} else if (open && !open.currentValue) {
+			if (this.trigger) {
+				this.trigger.focus();
+			}
+		}
+	}
```

Comment:
I don't totally want to early return in this case, if we end up needing more handlers in `ngOnChanges`

Reply:
Sure, in that case, how about
```suggestion
		if (open) {
			if (open.currentValue) {
				// `100` is just enough time to allow the modal
				// to become visible, so that we can set focus
				setTimeout(() => this.focusInitialElement(), 100);
			} else if (this.trigger) {
				this.trigger.focus();
			}
		}
	}
```

Reply:
Yeah that looks like it should work 👍